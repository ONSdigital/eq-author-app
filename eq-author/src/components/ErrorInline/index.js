import React from "react";
import PropTypes from "prop-types";
import Icon from "./icon-error.svg?inline";
import styled from "styled-components";
import { colors } from "constants/theme";
import VisuallyHidden from "components/VisuallyHidden";

const Container = styled.div`
  position: absolute;
  right: 0;
  bottom: -2em;
  color: ${colors.red};
  fill: ${colors.red};
  font-weight: normal;
`;

const IconContainer = styled.span`
  display: flex;
  align-items: center;
  line-height: 1.3;
`;

const ErrorInline = ({ children }) => {
  if (!children) {
    return null;
  }

  return (
    <Container data-test="error">
      <IconContainer>
        <Icon />
        <span role="alert">
          <VisuallyHidden>Error:&nbsp;</VisuallyHidden>
          {children}
        </span>
      </IconContainer>
    </Container>
  );
};

ErrorInline.propTypes = {
  children: PropTypes.string,
};

export default ErrorInline;
