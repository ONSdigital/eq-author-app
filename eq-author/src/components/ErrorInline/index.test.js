import React from "react";
import { shallow } from "enzyme";
import ErrorInline from ".";

describe("ErrorInline", () => {
  it("should render", () => {
    expect(shallow(<ErrorInline>Oops</ErrorInline>)).toMatchSnapshot();
  });
});
