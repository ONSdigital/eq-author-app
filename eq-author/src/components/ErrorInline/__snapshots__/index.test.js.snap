// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorInline should render 1`] = `
<ErrorInline__Container
  data-test="error"
>
  <ErrorInline__IconContainer>
    <Component />
    <span
      role="alert"
    >
      <VisuallyHidden>
        Error:Â 
      </VisuallyHidden>
      Oops
    </span>
  </ErrorInline__IconContainer>
</ErrorInline__Container>
`;
